### PR TITLE
Upgrade to use latest Go v1.15.2

### PR DESCRIPTION
**Pull request by nopcoder**
https://blog.golang.org/go1.15

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/treeverse/lakefs
 
-go 1.14
+go 1.15
```
Comment (arielshaqed) [inline go.mod:3]: Not sure about this.  I agree that we _should_ test also with bleading-edge Go.  But it is impractical to _require_ this version in order to compile.  I would feel better about continuing to use 1.14 for a while, and commit to support 1.15 and 1.14 by running GitHub testing actions in a matrix.
WDYT?

**Review by arielshaqed — CHANGES_REQUESTED** [review]
Blocking to decide on a strategy for Go version support.

**Comment by nopcoder** [comment]
I'll create an issue for document the strategy first

File: go.mod (hunk 1 above)
Comment (nopcoder) [reply go.mod:3]: No commit to supporting a specific version - if we will have issues with go 1.15 we can rollback the change.
Prefer to use what any contributor will get when he/she does 'brew install go'.
Addressing the current Go version in PR #674.
Can defer the upgrade but I don't have a good timeline on when it is the right time to switch, can go with always one version back - or only after X maintenance releases of the Go toolchain.

**Comment by arielshaqed** [comment]
After discussing it today f2f, let's go (_sorry_) with this -- Go is an internal dependency, and we can move forward. Thanks!

**Review by arielshaqed — APPROVED** [review]

**Pull request merged**